mapper/tedge_mapper/src/converter.rs

```diff
@@ -1,17 +1,43 @@
+use mqtt_client::{Message, Topic, TopicFilter};
+use std::fmt::Display;
+use tracing::error;
+
+#[derive(Debug)]
+pub struct MapperConfig {
+    pub in_topic_filter: TopicFilter,
+    pub out_topic: Topic,
+    pub errors_topic: Topic,
+}
+
 pub trait Converter: Send + Sync {
-    type Error;
+    type Error: Display;
 
-    fn convert(&self, input: &str) -> Result<String, Self::Error>;
+    fn get_mapper_config(&self) -> &MapperConfig;
 
-    fn convert_child_device_payload(
-        &self,
-        _input: &str,
-        _child_id: &str,
-    ) -> Result<String, Self::Error> {
-        Ok("".to_string())
+    fn get_in_topic_filter(&self) -> &TopicFilter {
+        &self.get_mapper_config().in_topic_filter
     }
 
-    fn convert_child_device_creation(&self, _child_id: &str) -> Option<mqtt_client::Message> {
-        None
+    fn convert_messages(&mut self, input: &Message) -> Result<Vec<Message>, Self::Error>;
+
+    fn convert(&mut self, input: &Message) -> Vec<Message> {
+        match self.convert_messages(input) {
+            Ok(messages) => messages,
+            Err(error) => {
+                error!("Mapping error: {}", error);
+                vec![Message::new(
+                    &self.get_mapper_config().errors_topic,
+                    error.to_string(),
+                )]
+            }
+        }
     }
 }
+
+pub fn make_valid_topic_or_panic(topic_name: &str) -> Topic {
+    Topic::new(topic_name).expect("Invalid topic name")
```

Comment:
```suggestion
    Topic::new(topic_name).expect("Invalid topic name: {}", topic_name)
```

Reply:
It requires `format!()`. I would skip them as we use those functions only to known topics, never panics a kind of.